/ RECOVER exception
recover exception
tests end
